src/schema/users.ts

```diff
@@ -567,6 +649,91 @@ export const resolvers: IResolvers<any, Context> = {
         )}.png?r=${randomStr}`,
       };
     },
+    updateUserProfile: async (
+      _,
+      { data, upload }: GQLUserParameters,
+      ctx,
+    ): Promise<GQLUser> => {
+      const repo = ctx.con.getRepository(User);
+      const user = await repo.findOne({ id: ctx.userId });
+
+      if (!user) {
+        throw new AuthenticationError(
+          JSON.stringify({ auth: 'User not found' }),
+        );
+      }
+
+      const isUsernameValid = new RegExp(/^@?(\w){1,15}$/).test(data.username);
+      const isGithubValid = new RegExp(/^@?([\w-]){1,39}$/i).test(data.github);
+      const isTwitterValid = new RegExp(/^@?(\w){1,15}$/).test(data.twitter);
+      const isHashnodeValid = new RegExp(/^@?([\w-]){1,39}$/i).test(
+        data.hashnode,
+      );
+
+      if (data.username && !isUsernameValid) {
+        throw new ValidationError(
+          JSON.stringify({ username: 'username is invalid' }),
+        );
+      }
+
+      if (data.github && !isGithubValid) {
+        throw new ValidationError(
+          JSON.stringify({ github: 'github is invalid' }),
+        );
+      }
+
+      if (data.twitter && !isTwitterValid) {
+        throw new ValidationError(
+          JSON.stringify({ twitter: 'twitter is invalid' }),
+        );
+      }
+
+      if (data.hashnode && !isHashnodeValid) {
+        throw new ValidationError(
+          JSON.stringify({ hashnode: 'hashnode is invalid' }),
+        );
+      }
+
+      const avatar = upload
+        ? await uploadAvatar(user.id, upload.createReadStream())
+        : user.image;
+
+      try {
+        const result = await ctx.con
+          .getRepository(User)
+          .save({ ...user, ...data, avatar });
```

Comment:
Make sure it works

```suggestion
          .save({ ...user, ...data, image: avatar });
```

Reply:
Yes. The only thing it gives back now is "must supply api_key" - which I think is available in prod environment. Or do we have to set it manually somewhere? Would love to be able to test it locally.